tomviz/PythonGeneratedDatasetReaction.cxx

```diff
@@ -541,31 +546,24 @@ void PythonGeneratedDatasetReaction::addDataset()
       generator.setArguments(args);
 
       const int shape[3] = { Nxy->value(), Nxy->value(), Nz->value() };
-      this->dataSourceAdded(generator.createDataSource(shape));
+      LoadDataReaction::dataSourceAdded(generator.createDataSource(shape), true,
+                                        false);
     }
   } // end of else if
 }
 
-void PythonGeneratedDatasetReaction::dataSourceAdded(
-  vtkSmartPointer<vtkSMSourceProxy> proxy)
-{
-  if (!proxy) {
-    return;
-  }
-  DataSource* dataSource = new DataSource(
-    proxy, DataSource::Volume);//, nullptr, DataSource::PersistenceState::Modified);
-  dataSource->setFileName(proxy->GetAnnotation(Attributes::LABEL));
-  LoadDataReaction::dataSourceAdded(dataSource, true /* default modules */,
-                                    false /* child */);
-}
-
-vtkSmartPointer<vtkSMSourceProxy>
-PythonGeneratedDatasetReaction::getSourceProxy(const QString& label,
-                                               const QString& script,
-                                               const int shape[3])
+DataSource* PythonGeneratedDatasetReaction::createGeneratedDataSource(
+  const QJsonObject& sourceInformation)
 {
-  PythonGeneratedDataSource generator(label);
-  generator.setScript(script);
+  PythonGeneratedDataSource generator(sourceInformation["label"].toString());
+  generator.setScript(sourceInformation["pythonScript"].toString());
```

Comment:
This should also be moved to script, I don't think we need to specify language, and if so would likely do so as another key-value pair.

Reply:
Makes sense